core/domain/collection_services.py

```diff
@@ -154,7 +184,18 @@ def get_collection_summary_from_model(collection_summary_model):
 
 
 def get_collection_by_id(collection_id, strict=True, version=None):
-    """Returns a domain object representing a collection."""
+    """Returns a domain object representing a collection.
+
+    Args:
+        collection_id: str. ID of the collection.
+        strict: bool. Indicates whether to fail noisily if no
+            collection with the given id exists in the datastore.
+        version: str. The version number of the collection to be retrieved.
```

Comment:
I traced the method CollectionModel.get to the one found in the class VersionedModel in `/core/storage/base_model/gae_models.py`, which ultimately calls `_get_snapshot_id` which treats the passed in `version` parameter as a string.